src/utils/extractTestResults.ts

```diff
@@ -4,13 +4,18 @@
 import { DynamoDBRecord } from 'aws-lambda';
 import { DynamoDB } from 'aws-sdk';
 import { TestActivity } from './testActivity';
+import { TestResultModel } from './testResult';
 
 export const extractBillableTestResults = (record: DynamoDBRecord): TestActivity[] => {
-  const data = DynamoDB.Converter.unmarshall(record.dynamodb.NewImage);
+  if (record.eventName === 'MODIFY' && isSameRecordDetails(record)) {
```

Comment:
Not particularly sure how expensive it is to run `DynamoDB.Converter.unmarshall` but for efficiencies sake, it might be worth nesting your conditional check on `isSameRecordDetails(record)` as otherwise it will perform both conditional checks for every event that comes in no matter whether it is REMOVE, INSERT, MODIFY. Nesting it will mean `isSameRecordDetails(record)` will only be ran for MODIFY events only and reduce the run time.

It might also be worth reusing your unmarshalled version of NewImage by assigning it to a variable before you call isSameRecordDetails, and changing the function to accept the unmarshalled versions of NewImage and OldImage. This means you can use that variable within the scope of `extractBillableTestResults` as well as `isSameRecordDetails`.

```suggestion
  const data = DynamoDB.Converter.unmarshall(record.dynamodb.NewImage) as TestResultModel;
  if (record.eventName === 'MODIFY') {
      const previousData = DynamoDB.Converter.unmarshall(record.dynamodb.NewImage) as TestResultModel;
      if (isSameRecordDetails(data, previousData)) return []
  }
  
  function isSameRecordDetails(newImage, oldImage): boolean {
  }
```

Reply:
In fact... ignore me about the nested ifs... realised `&&` will only evaluate the first condition if the first condition returns false